editorpythonbindings.md

```diff
@@ -0,0 +1,566 @@
+
+### The Attributes
+The Script Context has three new attributes to expose behavior for automation usage. The gem uses the Scope attribute to set behavior to usage in the Launcher, the Editor, or for Common.
+
+**Scope and ScopeFlags**
+```c++
+AZ::Script::Attributes::Scope
+AZ::Script::Attributes::ScopeFlags::Launcher
+AZ::Script::Attributes::ScopeFlags::Automation
+AZ::Script::Attributes::ScopeFlags::Common
+```
+
+The feature teams exposes either a Behavior Class or Behavior EBus by using the Script Scope attribute with either the Automation or Common flag.
+
+For example:
+```c++
+if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+{
+    behaviorContext->Class<MyType>("MyType")
+        ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
+        ->Attribute(AZ::Script::Attributes::Module, "types")
+        ;
+}
+```
+
+This exposes the MyType class to the behavior context (as a Behavior Class) with two attributes:
+
+    1. Scope with the Automation flag
+    2. Module with the text "types"
+
+
+This will expose the MyType class to the azlmbr.types module. It is possible to scope Behavior EBuses, Classes, constants, enums, and properties for Automation.
+
+A full example for a reflect function might look like this:
+
+```c++	
+void Reflect(AZ::ReflectContext* context)
+{
+    if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+    {
+        serializeContext->Class<MyType>()
+            ->Version(1)
+            ->Field("Data", &MyType::m_data)
+    }
+     
+    if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+    {
+        behaviorContext->Class<MyType>("MyType")
+            ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
+            ->Attribute(AZ::Script::Attributes::Module, "types")
+            ->Method("DoWork", &MyType::DoWork)
+              ->Attribute(AZ::Script::Attributes::Alias, "do_work")
+            ->Property("Data", &MyType::GetData, &MyType::SetData)
+              ->Attribute(AZ::Script::Attributes::Alias, "data")
+            ;
+    }
+}
+```
+
+##### Scoping For In Game Mode
+There are actually two separate game client modes 1) the game Launcher run time and 2) the game mode inside the Editor. The Python scripts will not work in the first client mode since the Python VM is not loaded in the Launcher.
+
+The second mode can run Python script code but it is not fully supported; so a feature team can use Python code in the Editor's game mode but not all behavior has been set up to work in this game mode.
+
+The default Scope value is AZ::Script::Attributes::ScopeFlags::Launcher  since the scripting contexts for Lua and Scripting Canvas where the first mechanisms for binding behavior.
```

Comment:
The AZ::Script::Attributes::ScopeFlags::Common and AZ::Script::Attributes::ScopeFlags::Automation flags should also be mentioned